test/e2e/podtimingcontroller/podtimingcontroller.go

```diff
@@ -0,0 +1,90 @@
+//go:build e2e
+// +build e2e
+
+package podtimingcontroller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+func SetupWithManager(mgr ctrl.Manager, log logr.Logger, artifactDir string) error {
+	r := &podTimingReconciler{
+		client: mgr.GetClient(),
+		podEnterPhaseDurationSeconds: prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
The duration a pod remains in a given phase monotonically increases (i.e. never decreases), correct? If so should this be a counter?